packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -378,7 +415,7 @@ export class WorkflowExecute {
 
 		// 7. Recreate Execution Stack
 		const { nodeExecutionStack, waitingExecution, waitingExecutionSource } =
-			recreateNodeExecutionStack(graph, new Set(startNodes), runData, pinData ?? {});
+			recreateNodeExecutionStack(graph, startNodes, runData, pinData ?? {});
```

Comment:
Unrelated change, but `startNodes` is already a `Set`, no need to wrap it again.